pkg/oci/containerd.go

```diff
@@ -314,11 +316,34 @@ func (c *Containerd) CopyLayer(ctx context.Context, dgst digest.Digest, dst io.W
 		return err
 	}
 	defer ra.Close()
-	_, err = io.Copy(dst, content.NewReader(ra))
-	if err != nil {
+
+	// Use a channel to signal the completion of the copy
+	done := make(chan error, 1)
+	// Start a goroutine to perform the copy
+	go func() {
```

Comment:
I do not understand why a new go routine is needed here, as we are still waiting for the copy to complete in the end. Is there something I am missing?

Reply:
yes, you are correct, reread the code and found there is already go routine in main.go:
`g.Go(func() error {
		if err := metricsSrv.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
			return err
		}
		return nil
	})`
Removed go routine.